Source/WebCore/platform/graphics/gstreamer/GStreamerRegistryScanner.cpp

```diff
@@ -721,5 +739,105 @@ GStreamerRegistryScanner::RegistryLookupResult GStreamerRegistryScanner::isConfi
+
+    if (factories.hasElementForMediaType(codecElement, "audio/x-alaw") && factories.hasElementForMediaType(rtpElement, "audio/x-alaw"))
+        capabilities.codecs.append({ .mimeType = "audio/PCMA"_s, .clockRate = 8000, .channels = 1, .sdpFmtpLine = emptyString() });
+}
+
+void GStreamerRegistryScanner::fillVideoRtpCapabilities(Configuration configuration, RTCRtpCapabilities& capabilities)
+{
+    if (!m_videoRtpExtensions)
+        m_videoRtpExtensions = probeRtpExtensions(m_allVideoRtpExtensions);
+    if (m_videoRtpExtensions)
+        capabilities.headerExtensions = copyToVector(*m_videoRtpExtensions);
+
+    auto codecElement = ElementFactories::Type::VideoDecoder;
+    auto rtpElement = ElementFactories::Type::RtpDepayloader;
+    if (configuration == Configuration::Encoding) {
+        codecElement = ElementFactories::Type::VideoEncoder;
+        rtpElement = ElementFactories::Type::RtpPayloader;
+    }
+
+    auto factories = ElementFactories({ codecElement, rtpElement });
+
+    if (factories.hasElementForMediaType(codecElement, "video/x-h264") && factories.hasElementForMediaType(rtpElement, "video/x-h264")) {
+        // Profile levels are hardcoded here for the time being. It might be a good idea to actually
+        // probe those on the selected encoder.
+        capabilities.codecs.append({ .mimeType = "video/H264"_s, .clockRate = 90000,
+            .sdpFmtpLine = "level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=640c1f"_s });
+        capabilities.codecs.append({ .mimeType = "video/H264"_s, .clockRate = 90000,
+            .sdpFmtpLine = "level-asymmetry-allowed=1;packetization-mode=1;profile-level-id=42e01f"_s });
+        capabilities.codecs.append({ .mimeType = "video/H264"_s, .clockRate = 90000,
+            .sdpFmtpLine = "level-asymmetry-allowed=1;packetization-mode=0;profile-level-id=640c1f"_s });
+        capabilities.codecs.append({ .mimeType = "video/H264"_s, .clockRate = 90000,
+            .sdpFmtpLine = "level-asymmetry-allowed=1;packetization-mode=0;profile-level-id=42e01f"_s });
+    }
+
+    // TODO: Probe for video/H265 capabilies.
+    // TODO: Probe for video/AV1 capabilies.
+
+    if (factories.hasElementForMediaType(codecElement, "video/x-vp8") && factories.hasElementForMediaType(rtpElement, "video/x-vp8"))
+        capabilities.codecs.append({ .mimeType = "video/VP8"_s, .clockRate = 90000 });
+
+    if (factories.hasElementForMediaType(codecElement, "video/x-vp9") && factories.hasElementForMediaType(rtpElement, "video/x-vp9")) {
+        // Profile levels are hardcoded here for the time being. It might be a good idea to actually
+        // probe those on the selected encoder.
+        capabilities.codecs.append({ .mimeType = "video/VP9"_s, .clockRate = 90000, .sdpFmtpLine = "profile-id=0"_s });
+        capabilities.codecs.append({ .mimeType = "video/VP9"_s, .clockRate = 90000, .sdpFmtpLine = "profile-id=2"_s });
+    }
+}
```

Comment:
But these capabilities would also depend of the available decoders and encoders in the gstreamer setup, isn't it?

Reply:
Yes, the scanner already handles that. Not sure what you think is missing here :)

Reply:
Oh. Then I would be worried about the available profiles. Does the scanner also check that?

Reply:
There is some support for checking H.264 profiles/levels but it's not hooked in the WebRTC capabilities probing yet, there is a TODO (or FIXME, whatever) about this.